### PR TITLE
DAOS-8408 test: Remove self.control_method and self.dmg check in Test…

### PR DESCRIPTION
**Pull request by shimizukko**
…Pool

Check if the dmg_command passed in to TestPool.__init__ is not None.
Then remove all the checks for self.control_method and self.dmg
for the methods that don't support API.

Tag - Test
sec_acl - security/pool_security_acl.py
aggregatebasic - aggregation/aggregation_basic.py
pool_evict_basic - pool/evict_test.py

Skip-unit-tests: true
Test-tag: sec_acl aggregatebasic pool_evict_basic
Signed-off-by: Makito Kano <makito.kano@intel.com>

### CODE REVIEW

File: src/tests/ftest/util/test_utils_pool.py [hunk 1]
```diff
@@ -84,7 +81,10 @@ def __init__(self, context, dmg_command, cb_handler=None,
         # to destroy the pool with UUID, set this to False, then call destroy().
         self.use_label = True
 
+        if dmg_command is None:
+            raise TypeError("dmg_command is None at TestPool init!")
```
Comment (phender) [inline src/tests/ftest/util/test_utils_pool.py:85]:
```suggestion
        if not isinstance(dmg_command, DmgCommand):
            raise TypeError("dmg_command is not a DmgCommand object: {}".format(type(dmg_command)))
```

File: src/tests/ftest/util/test_utils_pool.py [hunk 2]
```diff
@@ -255,39 +222,25 @@ def create(self):
...
+        data = self.dmg.pool_create(**kwargs)
+
+        if self.dmg.result.exit_status == 0:
+            # Convert the string of service replicas from the dmg command
+            # output into an ctype array for the DaosPool object using the
+            # same technique used in DaosPool.create().
+            service_replicas = [
+                int(value) for value in data["svc"].split(",")]
+            rank_t = ctypes.c_uint * len(service_replicas)
+            rank = rank_t(*list([svc for svc in service_replicas]))
```
[the remark this answers is not in the document]
Comment (phender) [reply src/tests/ftest/util/test_utils_pool.py:236]:
```suggestion
            rank = rank_t(*service_replicas)
```

**Review by phender — DISMISSED** [review]
There are some docstring typos, but the code changes are good.

**Review by dinghwah — DISMISSED** [review]
Script updates LGTM.
The test under pr-head shows failed, but the tests are passed, I am also seeing this on the other PR commits, if this is an common issue?

**Comment by sylviachanoiyee** [comment]
Can we run "pr daily_regression" on this pr since the change is in pool utils?

**Review by phender — APPROVED** [review]

**Comment by shimizukko** [comment]
Test Failure:
osa/osa_offline_reintegration.py - DAOS-8231

**Review by sylviachanoiyee — APPROVED** [review]

**Pull request merged**